### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by forslund**
#### Description
- Slight modernization (intent_file_handler -> intent_handler, etc)
- Improved docstrings
- Add more tests
- Add missing dialog files

#### Type of PR
- [ ] Bugfix
- [ ] Feature implementation
- [x] Refactor of code (without functional changes)
- [x] Documentation improvements
- [x] Test improvements

#### Testing
Check that the VK tests passes

### CODE REVIEW

**Comment by forslund** [comment]
I'll likely do a followup PR isolating the notification system. It should probably be isolated in it's own class.

File: test/behave/reminder.feature [hunk 1]
```diff
@@ -1,5 +1,8 @@
 Feature: mycroft-reminder
 
+  Background:
+    Given no reminders are set
```
Comment (krisgesling) [inline test/behave/reminder.feature:4]: I hadn't seen Background before, this is so useful!

We could also shift the "Given an english speaking user" into these

File: test/behave/reminder.feature [hunk 2]
```diff
@@ -11,3 +14,42 @@ Feature: mycroft-reminder
...
+
+  Scenario Outline: Reminder without time
+    Given an english speaking user
+     When the user says "remind me to be awesome"
+     Then "mycroft-reminder" should reply with dialog from "ParticularTime.dialog"
+     And the user says "<reminder tomorrow>"
+     And "mycroft-reminder" should reply with dialog from "SavingReminderTomorrow.dialog"
+
+  Examples:
+    | reminder tomorrow |
```
Comment (krisgesling) [inline test/behave/reminder.feature:27]: Could we call this something like "a specific time in the future"?

File: test/behave/reminder.feature [hunk 3]
```diff
@@ -11,3 +14,42 @@ Feature: mycroft-reminder
...
+
+  Scenario Outline: Reminder without time, Negative
+    Given an english speaking user
+     When the user says "remind me to be awesome"
+     Then "mycroft-reminder" should reply with dialog from "ParticularTime.dialog"
+     And the user says "<no thanks>"
+     And "mycroft-reminder" should reply with dialog from "Ok.dialog"
+
+  Examples:
+    | no thanks |
```
Comment (krisgesling) [inline test/behave/reminder.feature:39]: maybe just "no"

File: __init__.py [hunk 4]
```diff
@@ -248,12 +248,12 @@ def add_unspecified_reminder(self, msg=None):
             return self.add_new_reminder(msg)
 
         response = self.get_response('ParticularTime')
-        if response and not self.response_is_negative(response):
-            # Check if a time was also in the response
-            dt, rest = extract_datetime(response) or (None, None)
+        # Check if a time was in the response
+        dt, rest = extract_datetime(response) or (None, None)
+        if dt or self.response_is_affirmative(response):
```
Comment (krisgesling) [inline __init__.py:281]: Took me a minute to work this one out but I definitely like the change.

**Review by krisgesling — APPROVED** [review]
This is excellent, thanks Ake.

A few minor suggestions on how we label Examples columns in VK tests but they really aren't of any material significance so I'm going to merge this as is.

**Pull request merged**